src/signal.c

```diff
@@ -0,0 +1,23 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <signal.h>
+#include "signal.h"
+
+void jq_handle_sigpipe(int signum)
+{
+#ifndef WIN32
+  signal(SIGPIPE, SIG_IGN);
+  fprintf(stderr, "jq: error: received SIGPIPE\n");
+  exit(128 + SIGPIPE);
```

Comment:
`exit()` is not async-signal-safe, so it can't be used from a signal handler.  Use `_exit()` instead.

Reply:
If and when we add an FFI this (exiting in a signal handler) will probably not be a safe thing to do.